spec/13-syntax-summary.md

```diff
@@ -21,7 +21,7 @@ digit            ::=  ‘0’ | … | ‘9’
 paren            ::=  ‘(’ | ‘)’ | ‘[’ | ‘]’ | ‘{’ | ‘}’
 delim            ::=  ‘`’ | ‘'’ | ‘"’ | ‘.’ | ‘;’ | ‘,’
 opchar           ::=  // printableChar not matched by (whiteSpace | upper | lower |
-                      // letter | digit | paren | delim | Unicode_Sm | Unicode_So)
+                      // letter | digit | paren | delim)
```

Comment:
It seems this was a typo in https://github.com/scala/scala/commit/5135bae5a7. `Unicode_Sm` and `Unicode_So` should be included in `opchar`, not excluded:

```
scala> def ☏ = "ring ring"
def $u260F: String
```

There should also be some explanation what `Unicode_Sm` / `Unicode_So` are (took me a while; it's not https://www.fileformat.info/info/unicode/char/2120/index.htm. It's https://www.compart.com/en/unicode/category).

`upper | lower | letter` is redundant, could be just `letter`

I think we should just enumerate the allowed characters instead of narrowing down from `printableChar`. That would also be closer to the implementation:

https://github.com/scala/scala/blob/107335443da97ad745094bfad69dd8fcf042c30f/src/compiler/scala/tools/nsc/ast/parser/Scanners.scala#L844-L866

Not sure why one of these lists includes `/`, but not the other. `isSpecial` is `Unicode_Sm | Unicode_So`.

I got a bit more confused by `:`, since `def f_: = 1` is allowed but `def : = 1` is a syntax error, I guess because `FunDef`. Similar for `@`.

Reply:
Sorry, I didn't notice the PR, so I commented on [the ticket](https://github.com/scala/bug/issues/12260#issuecomment-853137334). I reminded us there that the text in chapter one should be kept in sync with the summary, or vice-versa.

`:` is a reserved word.

I looked at [wikipedia](https://en.wikipedia.org/wiki/Mathematical_operators_and_symbols_in_Unicode) for a symbol to cut/paste in my comment on the ticket. Someone asked on twitter what is the seminal moment for modern mathematics, and they answered typesetting with nice symbols.